Src/Sarif.PatternMatcher.Cli/AnalyzeDatabaseCommand.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Text;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+using Kusto.Data.Common;
+using Kusto.Data.Net.Client;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Cli
+{
+    internal class AnalyzeDatabaseCommand : CommandBase
+    {
+        private const int DefaultBatchSize = 10000;
+        private readonly HashSet<string> disabledSkimmers;
+        private readonly ConcurrentDictionary<string, string> content;
+        private bool dynamicValidation;
+        private Channel<string> contentChannel;
+
+        public AnalyzeDatabaseCommand()
+        {
+            disabledSkimmers = new HashSet<string>();
+            content = new ConcurrentDictionary<string, string>();
+        }
+
+        public int Run(AnalyzeDatabaseOptions options)
+        {
+            try
+            {
+                options.Threads = options.Threads > 0 ? options.Threads : Environment.ProcessorCount * 4;
+                dynamicValidation = options.DynamicValidation;
+
+                var channelOptions = new BoundedChannelOptions(int.MaxValue)
```

Comment:
>int.MaxValue [](http://example.com/codeflow?start=63&length=12)

the speed of things being added >> processing rate. Based on that, we can not have a fixed small size.